person.rb

```diff
@@ -0,0 +1,77 @@
+class Nameable
+  attr_accessor :name
+
+  def initialize(name)
+    @name = name
+  end
+
+  def correct_name
+    raise NotImplementedError, 'Kindly Implement correct_name method to return the correct name'
+  end
+end
+
+class Person < Nameable
+  attr_accessor :name, :age
+
+  def initialize(age, name)
+    super(name)
+
+    @name = name
+
+    @age = age
+  end
+
+  def can_use_services?
+    if is_of_age? || @parent_permission
+      true
+    else
+      false
+    end
+  end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
+
+    @nameable = nameable
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name[0..9]
+    else
+      @nameable.correct_name
+    end
+  end
+end
```

Comment:
- [ ] It would be better to separate the three classes in this file to different files this will make your code more readable 👍🏿

Reply:
😄 Though I think its not necessary to differentiate small classes into different files. But In this case I did approved your requirements. 
Thanks
Happy Coding 😄